flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1373,6 +1371,7 @@ private void cleanupAfterCompletedCheckpoint(
                     completedCheckpoint.getTimestamp(),
                     extractIdIfDiscardedOnSubsumed(lastSubsumed));
         }
+        pendingCheckpoint.getCompletionFuture().complete(completedCheckpoint);
```

Comment:
I prefer to do this right in ahead of the invocation of `reportCompletedCheckpoint(...)`. So that we report completed checkpoints only after it completes.
Do you have any concern for it?